tests/e2e/utils/node_utils.go

```diff
@@ -50,7 +51,7 @@ func GetAgentNodes(cs clientset.Interface) ([]v1.Node, error) {
 	}
 	ret := make([]v1.Node, 0)
 	for _, node := range nodesList.Items {
-		if !isMasterNode(&node) {
+		if !isMasterNode(&node) && !isVirtualKubeletNode(&node) {
```

Comment:
Need to ignore virtual kubelet node or else it will create a deployment with 4033 pods and cause the test to fail.
```
I0205 16:42:06.395] Feb  5 15:49:43.350: INFO: will create 4033 pod, each 200m size
```

Reply:
Good catch for this.